jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/ResourceFactory.java

```diff
@@ -446,6 +446,36 @@ default Resource newJarFileResource(URI uri)
         return newResource(URIUtil.toJarFileUri(uri));
     }
 
+    /**
+     * Convert an anonymous Object to a Resource.
+     *
+     * <p>
+     *     The supported types: {@link Path}, {@link String}, {@link URI}, {@link URL}, and {@link Resource}
+     * </p>
+     *
+     * @param obj if the object to convert
+     * @return the Resource representing the provided obj
+     * @throws IllegalArgumentException if the object is none of the supported types.
+     */
+    default Resource asResource(Object obj)
+    {
+        if (obj == null)
+            return null;
+
+        if (obj instanceof Path path)
+            return newResource(path);
+        if (obj instanceof String str)
+            return newResource(str);
+        if (obj instanceof URI uri)
+            return newResource(uri);
+        if (obj instanceof URL url)
+            return newResource(url);
+        if (obj instanceof Resource res)
+            return res;
+
+        throw new IllegalArgumentException("Cannot convert %s to a Resource".formatted(obj.getClass().getName()));
```

Comment:
The alternative is to do
```suggestion
        return newResource(obj.toString());
```

Reply:
That has the possibility of returning null with no error.
That seems less useful than an IAE.

Reply:
I don't think so.  The contract of `newResource(String)` is to throw ISE on an issue and to never return null.
The difference is that my suggestion would allow an unforeseen type to be set as an attribute that would work with this method, so long as it's toString returned a valid resource string.
So for now, we still get the IAE, but we are more future proof.